yaml-tests/src/test/java/MultiServerConnectionFactoryTest.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * MultiServerFactoryTest.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2025 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import com.apple.foundationdb.relational.api.RelationalConnection;
+import com.apple.foundationdb.relational.yamltests.MultiServerConnectionFactory;
+import com.apple.foundationdb.relational.yamltests.YamlRunner;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+
+import javax.annotation.Nonnull;
+import java.net.URI;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.Set;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class MultiServerConnectionFactoryTest {
+    @ParameterizedTest
+    @CsvSource({"0", "1", "2", "3", "4"})
+    void testDefaultPolicy(String initialConnection) throws SQLException {
+        int ic = Integer.parseInt(initialConnection);
+        MultiServerConnectionFactory classUnderTest = new MultiServerConnectionFactory(
+                MultiServerConnectionFactory.ConnectionSelectionPolicy.DEFAULT,
+                ic,
+                dummyConnectionFactory(),
+                List.of(dummyConnectionFactory()));
+        assertEquals(ic, classUnderTest.getCurrentConnection());
+        classUnderTest.getNewConnection(URI.create("Blah"));
+        assertEquals(ic, classUnderTest.getCurrentConnection());
+        classUnderTest.getNewConnection(URI.create("Blah"));
+        assertEquals(ic, classUnderTest.getCurrentConnection());
+    }
+
+    @ParameterizedTest
+    @CsvSource({"0", "1", "2", "3", "4"})
+    void testAlternatePolicy(String initialConnection) throws SQLException {
+        int ic = Integer.parseInt(initialConnection);
+        MultiServerConnectionFactory classUnderTest = new MultiServerConnectionFactory(
+                MultiServerConnectionFactory.ConnectionSelectionPolicy.ALTERNATE,
+                ic,
+                dummyConnectionFactory(),
+                List.of(dummyConnectionFactory()));
+        assertEquals(ic, classUnderTest.getCurrentConnection());
+        classUnderTest.getNewConnection(URI.create("Blah"));
+        assertEquals(ic + 1, classUnderTest.getCurrentConnection());
+        classUnderTest.getNewConnection(URI.create("Blah"));
+        assertEquals(ic + 2, classUnderTest.getCurrentConnection());
```

Comment:
I wonder whether it would make sense to try and have a more integrated test, similar to `SupportedVersionTest`, where the test would fail if it did anything other than alternate.

Reply:
Not sure what you mean. Does the added coverage meet that?

Reply:
I was thinking something that has an actual `.yamsql` file, and would fail if it did anything other than alternate connections. But I think that would be a lot of work, and not a lot of additional coverage.

Reply:
I see. That's a good idea, but like you said, may take some time. I'll keep it around for future improvements?